build-tools/packages/build-cli/docs/typetestDetails.md

```diff
@@ -0,0 +1,153 @@
+run for each package. You can run only one part at a time using the `--prepare` and `--generate` flags.
+
+[api type validation]: https://github.com/microsoft/FluidFramework/wiki/API-Type-Validation
+
+## Preparing package.json
+
+_Preparing package.json_ determines the baseline previous version to use, then sets that version in package.json. If the
+previous version changes after running preparation, then `npm install` must be run before the generate step will run
+correctly.
+
+Optionally, any type tests that are marked "broken" in package.json can be reset using the `--reset` flag during
+preparation. This is useful when resetting the type tests to a clean state, such as after a major release.
+
+## Generating tests
+
+Generating test modules takes the type test information from package.json, most notably any known broken type tests, and
+generates test files that should be committed. By default, the generated files will contain `.generated` in their name,
+but this can be suppressed with the `--no-generateInName` flag.
+
+## Branch configuration
+
+Type tests can be configured to use different baseline versions on a given branch depending on the type of release that
+the branch is designated for. For example, for the client release group, the _next_ branch is the _major version series
+branch_ and _main_ is the _minor version series branch_. This can be declared in the root package.json, in the
+`fluidBuild.repoPackages` section. For example, the following configuration designates the _main_ and _lts_ branches as
+minor version series branches, while the _next_ branch is designated for major releases.
+
+```json
+"fluidBuild": {
+  "repoPackages": {
+    "client": {
+      "directory": "",
+      "ignoredDirs": [],
+      "branchReleaseTypes": {
+        "main": "minor",
+        "lts": "minor",
+        "release/**": "patch",
+        "next": "major"
+      }
+    }
+  }
+}
+```
+
+The branch names can be globs. They are matched using [minimatch](https://www.npmjs.com/package/minimatch).
+
+The type test generator takes this information into account when calculating the baseline version to use when it's run
+on a particular branch. Baseline versions are set as follows based on the branch release designation:
+
+| Branch release designation | Baseline version | Example: version 2.3.4 |
+| -------------------------- | ---------------- | ---------------------- |
+| `patch`                    | `previousPatch`  | **2.3.3**              |
+| `minor`                    | `^previousMinor` | **^2.2.0**             |
+| `major`                    | `^previousMajor` | **^1.0.0**             |
+
+### Configuring a branch for a specific baseline
+
+It may be useful to configure a branch for a specific baseline instead of the default based on the branch release
+designation. To do this, you can use any of the following strings instead of major/minor/patch.
+
+- `baseMajor`
```

Comment:
Maybe some super-short description here on "base" would be helpful. When I think of "previous" I think about previous to existing branch version, but "base" I cannot fit into my mental model :)

Reply:
Clarified both here and in the tsdocs:

The "base" versions are calculated by zeroing out all version segments lower than the base. That is, for a version v,
the baseMajor version is `${v.major}.0.0` and the baseMinor version is `${v.major}.${v.minor}.0`.

The "previous" versions work similarly, but the major/minor/patch segment is reduced by 1. That is, for a version v,
the previousMajor version is `${min(v.major - 1, 1)}.0.0`, the previousMinor version is
`${v.major}.${min(v.minor - 1, 0)}.0`, and the previousPatch is `${v.major}.${v.minor}.${min(v.patch - 1, 0)}.0`.

The "previous" versions never roll back below 1 for the major version and 0 for minor and patch. That is, the
previousMajor, previousMinor, and previousPatch versions for `1.0.0` are all `1.0.0`.